### PR TITLE
added device-name component to the dashboard and extended one-shot-de…

### PR DESCRIPTION
**Pull request by JuriBog**
…vice/create endpoint with a device name paramter

### CODE REVIEW

File: dashboard/src/components/dashboard.svelte [hunk 1]
```diff
@@ -39,17 +42,17 @@
...
 			content: [
 				{
 					icon: 'person-badge',
 					boxColor: 'transparent',
-					value: device.id ?? '-'
+					value: device.name ?? '-'
 				},
 				{
-					value: device.channelAddress ?? '-'
+					value: `channel-${device.name}` ?? '-'
```
Comment (mastrogiovanni) [inline dashboard/src/components/dashboard.svelte:55]: In reality this information don't need to be inferred. What happens if the backend logic in giving the names will change? This information need to be provided by the backend together with the name.

File: dashboard/src/lib/index.ts [hunk 2]
```diff
@@ -47,16 +47,16 @@ export async function getDevices(creatorId: string): Promise<Device[]> {
...
 }
 
-export async function createDevice(): Promise<void> {
+export async function createDevice(deviceName: string): Promise<void> {
     try {
         // Create a channel
         // Timeout to see progress bar animation (from 0 to 0.33)
         setTimeout(() => deviceCreationProgress.set(0.33), 100)
-        const channel = await createChannel([{ type: 'cxc', source: 'cxc' }]);
+        const channel = await createChannel(`channel-${deviceName}`,'',[{ type: 'cxc', source: 'cxc' }]);
```
Comment (mastrogiovanni) [inline dashboard/src/lib/index.ts:55]: the logic of composition of channelName need to be in a separate method is used in more than one place.

**Review by mastrogiovanni — CHANGES_REQUESTED** [review]
Few small changes mostly related to the creation of a single point to establish channelName from the deviceName

File: one-shot-device/src/devices/device-registration.controller.ts [hunk 3]
```diff
@@ -5,10 +5,10 @@ import { ApiTags, ApiOperation, ApiResponse, ApiOkResponse, ApiNotFoundResponse,
 @ApiTags('one-shot-device')
 @Controller('/api/v1/one-shot-device')
 export class DeviceRegistrationController {
-	constructor(private readonly deviceRegistrationService: DeviceRegistrationService) {}
+	constructor(private readonly deviceRegistrationService: DeviceRegistrationService) { }
 	private readonly logger: Logger = new Logger(DeviceRegistrationController.name);
 
-	@Post('/create/:channelAddress/:creator')
+	@Post('/create/:channelAddress/:creator/:deviceName')
```
Comment (dominic22) [inline one-shot-device/src/devices/device-registration.controller.ts:11]: the deviceName should be part of the body not path

**Review by jacekroszkowiakdev — APPROVED** [review]
Looks all ok to me - good job!

File: dashboard/src/lib/index.ts [hunk 4]
```diff
@@ -56,8 +56,12 @@ export async function createDevice(deviceName: string): Promise<void> {
         // Create a device
         if (channel) {
             deviceCreationProgress.set(0.66)
-            const deviceResponse = await fetch(`${SLU_API_BASE_URL}/one-shot-device/create/${channel?.channelAddress}/${get(authenticationData)?.did}/${deviceName}`, {
+            const deviceResponse = await fetch(`http://localhost:3010/api/v1/one-shot-device/create/${channel?.channelAddress}/${get(authenticationData)?.did}`, {
```
Comment (dominic22) [inline dashboard/src/lib/index.ts:59]: why is there a static url?

File: one-shot-device/src/devices/device-registration.controller.ts [hunk 5]
```diff
@@ -17,10 +17,21 @@ export class DeviceRegistrationController {
...
+				description: 'For creating one device with a name.',
+				value: {
+					name: 'device-name'
+				}
+			}
+		}
+	})
 	@ApiOperation({ summary: 'Create a channel as manager' })
-	async createAndSubscribe(@Param('channelAddress') channelAddress: string, @Param('creator') creator: string, @Param('deviceName') deviceName: string) {
+	async createAndSubscribe(@Param('channelAddress') channelAddress: string, @Param('creator') creator: string, @Body() deviceName: { name: string }) {
```
Comment (dominic22) [inline one-shot-device/src/devices/device-registration.controller.ts:32]: maybe better to call the body `body` and not deviceName

Comment (dominic22) [reply one-shot-device/src/devices/device-registration.controller.ts:32]: is this name now mandatory or optional?

**Review by dominic22 — APPROVED** [review]

**Pull request merged**